### PR TITLE
adds composer support in README and adds autoload back

### PR DESCRIPTION
**Pull request by bshaffer**
This is to address [user concerns](http://stackoverflow.com/questions/33483326/how-do-i-use-composer-to-setup-google-api-php-client) regarding composer support. 

In addition, we will be adding a downloadable zip file to each project that includes all the dependencies, for those users who would prefer to not use composer.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +16,34 @@ http://developers.google.com/api-client-library/php
...
+
+### Composer
+
+Install this library via [composer](https://getcomposer.org) by running the following command:
+
+```sh
+# change directories to the root of your project
+cd /path/to/my-project
+# if you don't already have composer, install it
+curl -sS https://getcomposer.org/installer | php
```
Comment (tmatsuo) [inline README.md]: I would defer it to the guide on the composer website.

File: README.md [hunk 2]
```diff
@@ -16,7 +16,34 @@ http://developers.google.com/api-client-library/php
...
+# install this library
+./composer.phar require google/apiclient
```
Comment (tmatsuo) [inline README.md]: There are multiple way to install composer and the command line varies. How about just 

```sh
$ composer require google/apiclient
```
?

File: README.md [hunk 3]
```diff
@@ -16,7 +16,34 @@ http://developers.google.com/api-client-library/php
...
+
+> The above example adds a file `composer.phar` to your project working directory for simplicity. We recommend
+> you [install composer globally](https://getcomposer.org/doc/00-intro.md#globally) for local development.
+
+### Autoloading
+
+Once installed, be sure to include the autoloader:
+
+```php
+require_once '/path/to/google-api-php-client/vendor/autoload.php';
```
Comment (tmatsuo) [inline README.md]: Is the path `/path/to/google-api-php-client` correct?

Comment (tmatsuo) [inline README.md]: I would say `/path/to/your-project`

File: README.md (hunk 1 above)
Comment (tmatsuo) [inline README.md]: Here is the reaosn why.

Well, curling a script from the web and pipe it into an interpreter, might be a trend, but in my opinion, it's dangerous in some senses. When composer website started to suggest more secure way, this document will provide an orphaned, insecure instruction.

Comment (bshaffer) [inline README.md]: I am really trying to keep this guide simple. As the stack overflow thread shows, some users are entirely unfamiliar with composer. Sending them to the website for installation will just increase the bounce rate. That is why I went this way... to get them up and going the fastest.

File: README.md (hunk 2 above)
Comment (bshaffer) [inline README.md]: that only works if they've taken the time to install composer globally. If they've done that, I assume they know the difference (and I mention it in the quote below)

File: README.md (hunk 3 above)
Comment (bshaffer) [inline README.md]: yeah, hmm... I mean't it specifically for those who downloaded the release w/o composer. But you're right, it's confusing.

File: README.md (hunk 1 above)
Comment (tmatsuo) [inline README.md]: Well, people who don't know composer can still download the zip file, right?

Comment (bshaffer) [inline README.md]: Yes, but composer should still be encouraged, hence the example.

Comment (tmatsuo) [inline README.md]: On https://getcomposer.org/doc/00-intro.md
There is a mention about Windows installer, which is better than the command line we are showing. Does the command line work with Windows console?

Comment (bshaffer) [inline README.md]: no clue, but I think you're right... pointing to the composer website is a better long term solution, and will make for a cleaner readme

**Comment by bshaffer** [comment]
@tmatsuo PTAL

File: src/Google/autoload.php [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+$file = __DIR__ . '/../../vendor/autoload.php';
```
Comment (kesar) [inline src/Google/autoload.php]: vendor could have a different folder name ( https://getcomposer.org/doc/06-config.md#vendor-dir )

Comment (tmatsuo) [inline src/Google/autoload.php]: Good point :)
Can you use __DIR__ . '/../autoload.php' instead?

Comment (bshaffer) [inline src/Google/autoload.php]: @kesar this is a deprecated file meant to move users of v1.x over to 2 with minimal difficulty. If a user has a custom `vendor` directory, they should not be using this file anyway, so not think it's worth handling this edge case.

@tmatsuo I am unclear what that would accomplish.

Comment (tmatsuo) [inline src/Google/autoload.php]: > I am unsure what that would accomplish.

@bshaffer I think it should work even you use custom `vendor` directory. I may be wrong though.

Comment (tmatsuo) [inline src/Google/autoload.php]: Ah, OK. I misunderstand that `src/Google/autoload.php` is installed when you're using composer.

Comment (tmatsuo) [inline src/Google/autoload.php]: Alright. LGTM

**Pull request merged**